Backend/Backend/Services/EpisodeService.cs

```diff
@@ -81,12 +128,19 @@ public async Task<bool> DeleteEpisode(Episode episode, DeleteEpisodeRequest dele
 
                         }
 
-                        bool? editted = await  _fileService.EditFile(episode.AudioFile!, editEpisodeRequest.AudioFile);
-                        if(editted == false)
+                        bool? editted =  _fileService.EditFile(episode.AudioFile!, editEpisodeRequest.AudioFile);
+                        if(editted == true)
                         {
-                            throw new Exception("Failed to Edit the file");
+                            episode.AudioFile!.Name = editEpisodeRequest.AudioFile.FileName;
```

Comment:
you should've probably kept the condition the way it was beforehand. This would let you keep the exception there, and remove the else statement.